registry/servicediscovery/store/cache_manager.go

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package store
+
+import (
+	"encoding/gob"
+	"os"
+	"sync"
+	"time"
+)
+
+import (
+	"github.com/dubbogo/gost/log/logger"
+
+	"github.com/hashicorp/golang-lru"
+)
+
+type CacheManager struct {
+	name         string        // The name of the cache manager
+	cacheFile    string        // The file path where the cache is stored
+	cacheExpired time.Duration // The duration after which the cache expires
+	stop         chan struct{} // Channel used to stop the cache expiration routine
+	cache        *lru.Cache    // The LRU cache implementation
+	lock         sync.Mutex
+}
+
+type Item struct {
+	Key   string
+	Value interface{}
+}
+
+// NewCacheManager creates a new CacheManager instance.
+// It initializes the cache manager with the provided parameters and starts a routine for cache expiration.
+func NewCacheManager(name, cacheFile string, cacheExpired time.Duration, maxCacheSize int) (*CacheManager, error) {
+	cm := &CacheManager{
+		name:         name,
+		cacheFile:    cacheFile,
+		cacheExpired: cacheExpired,
+		stop:         make(chan struct{}),
+	}
+	cm.cache, _ = lru.New(maxCacheSize)
```

Comment:
Please do not ingore the error, although the current implemented lru returns `nil`.